src/overlays/actors/ovl_En_Sob1/z_en_sob1.c

```diff
@@ -21,164 +85,1636 @@ const ActorInit En_Sob1_InitVars = {
+    if (this->shopItemSelectedTween <= 0.15f) {
+        this->shopItemSelectedTween = 0.0f;
+    }
+    EnSob1_PositionSelectedItem(this);
+    if (this->shopItemSelectedTween == 0.0f) {
+        return true;
+    }
+    return false;
+}
+
+void EnSob1_UpdateItemSelectedProperty(EnSob1* this) {
+    EnGirlA** items = this->items;
+    EnGirlA* item;
+    s32 i;
+
+    for (i = 0; i < ARRAY_COUNT(this->items); i++, items++) {
+        item = *items;
+        if (item != NULL) {
+            if (this->actionFunc != EnSob1_SelectItem && this->actionFunc != EnSob1_CannotBuy &&
+                this->drawCursor == 0) {
+                item->isSelected = false;
+            } else {
+                item->isSelected = this->cursorIdx == i ? true : false;
+            }
+        }
+    }
+}
+
+void EnSob1_UpdateCursorAnim(EnSob1* this) {
+    f32 t;
+
+    t = this->cursorAnimTween;
+    if (this->cursorAnimState == 0) {
+        t += 0.05f;
+        if (t >= 1.0f) {
+            t = 1.0f;
+            this->cursorAnimState = 1;
+        }
+    } else {
+        t -= 0.05f;
+        if (t <= 0.0f) {
+            t = 0.0f;
+            this->cursorAnimState = 0;
+        }
+    }
+    this->cursorColorR = COL_CHAN_MIX(0, 0.0f, t);
+    this->cursorColorG = COL_CHAN_MIX(80, 80.0f, t);
+    this->cursorColorB = COL_CHAN_MIX(255, 0.0f, t);
+    this->cursorColorA = COL_CHAN_MIX(255, 0.0f, t);
+    this->cursorAnimTween = t;
+}
+
+void EnSob1_UpdateStickDirectionPromptAnim(EnSob1* this) {
+    f32 arrowAnimTween = this->arrowAnimTween;
+    f32 stickAnimTween = this->stickAnimTween;
+
+    // Possbily fake temps
+    s32 maxColor = 255;
+    f32 tmp;
+
+    if (this->arrowAnimState == 0) {
+        arrowAnimTween += 0.05f;
+        if (arrowAnimTween > 1.0f) {
+            arrowAnimTween = 1.0f;
+            this->arrowAnimState = 1;
+        }
+    } else {
+        arrowAnimTween -= 0.05f;
+        if (arrowAnimTween < 0.0f) {
+            arrowAnimTween = 0.0f;
+            this->arrowAnimState = 0;
+        }
+    }
+    this->arrowAnimTween = arrowAnimTween;
+
+    if (this->stickAnimState == 0) {
+        stickAnimTween += 0.1f;
+        if (stickAnimTween > 1.0f) {
+            stickAnimTween = 1.0f;
+            this->stickAnimState = 1;
+        }
+    } else {
+        stickAnimTween = 0.0f;
+        this->stickAnimState = 0;
+    }
+
+    tmp = 155.0f * arrowAnimTween;
+
+    this->stickAnimTween = stickAnimTween;
+
+    this->stickLeftPrompt.arrowColorR = COL_CHAN_MIX(255, 155.0f, arrowAnimTween);
+    this->stickLeftPrompt.arrowColorG = COL_CHAN_MIX(maxColor, 155.0f, arrowAnimTween);
+    this->stickLeftPrompt.arrowColorB = COL_CHAN_MIX(0, -100, arrowAnimTween);
+    this->stickLeftPrompt.arrowColorA = COL_CHAN_MIX(200, 50.0f, arrowAnimTween);
+
+    this->stickRightPrompt.arrowColorR = (maxColor - ((s32)tmp)) & 0xFF;
+    this->stickRightPrompt.arrowColorG = (255 - ((s32)tmp)) & 0xFF;
```

Comment:
```suggestion
    this->stickRightPrompt.arrowColorR = (u8)(maxColor - ((s32)tmp));
    this->stickRightPrompt.arrowColorG = (u8)(255 - ((s32)tmp));
```
Would this still work?

Reply:
In its current state that does not help. I hope to eventually get the function into a state that these can also use the COL_CHAN_MIX, assuming that's possible.